File: simple_linear_regression.R
#Simple Linear Regression 

#Data preprocessing 

#Imoporting the dataset 
dataset = read.csv('Salary_Data.csv')
#dataset = dataset[, 2:3]

#splitting the dataset into the training set and test set
#библиотека для сплита - CaTools
#install.packages('caTools')
#устанавливаем рандомное число для сплита
set.seed(123)

#в питоне нам надо было указывать и матрицу Х и зависимый ветор У, здесь 
#нам надо указать только столбец зависимой переменной 
#в splitrartio мы просто указываем, процент набора данных для тренировки
split = sample.split(dataset$Salary, SplitRatio = 2/3)
#после выполнения строчки сверху, мы получим значение split, которое будет состоять из тру и фолс 
#тру - попадание в трейнинг сет, фолс - в тест сет 

#теперь нам надо отдельно создать тест сет и трейнинг сет
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

#Fitting Simple Linear Regression to the Training Set 
#formula = зависимая переменная, выраженная, как линейная комбинация независимой переменной 
#в formula мы указали, что зарплата пропорциональная годам опыта
#P-value показывает насколько большое влияние независимая переменная оказывает на зависимую переменную 
# (чем меньше Р-значением, тем больше влияния оказывает независимая переменная)

#Р-значение может пригтдиться при выборе из нескольких независимых переменных
regressor = lm(formula = Salary ~ YearsExperience,
               data = training_set)

#Predicting the test set results 
#Y_pred - вектор предсказаний 
#в предикт мы указываем нашу модель и переменную - new_data - те данные, которые мы будем предсказывать
y_pred = predict(regressor, newdata = test_set)

#Visualising the Training set results 
#install.packages('ggplot2')

ggplot()+
  geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary),
             colour = 'red') +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
            color = 'blue') +
  ggtitle('Salary VS Experience (Training set)') +
  xlab('Years of Experience') +
  ylab('Salary')

#Visualising the Test set results
ggplot()+
  geom_point(aes(x = test_set$YearsExperience, y = test_set$Salary),
             colour = 'red') +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
            color = 'blue') +
  ggtitle('Salary VS Experience (Test set)') +
  xlab('Years of Experience') +
  ylab('Salary')


  

File: simple_linear_regression.py
#Data Preprocessing

#Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd


#Importing dataset
dataset = pd.read_csv('Salary_Data.csv')

X = dataset.iloc[:, 0].values
Y = dataset.iloc[:, 1].values

#Splitting dataset to training set and test set 
from sklearn.model_selection import train_test_split
X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size = 1/3, random_state = 0)

#feature scaling 
"""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test) """

X_train = X_train.reshape(-1, 1)

Y_train = Y_train.reshape(-1, 1)

X_test = X_test.reshape(-1, 1)

#Fitting Simple Linear Regression to the Training Set 
#используем библиотеку linear_model 
from sklearn.linear_model import LinearRegression
#создаем модель линейной регрессии
regressor = LinearRegression()
#применяем модель к training_set, обучаем машину на наших примерах
#при использование fit ниже наша модель изучает зависимости между Х и У
regressor.fit(X_train, Y_train)

#Predicting the test set results 
#y_pred - вектор предсказаний зависимой переменной 
y_pred = regressor.predict(X_test)

#Visualising the training set 
plt.scatter(X_train, Y_train, color = 'red' )
plt.plot(X_train, regressor.predict(X_train), color = 'green' )
plt.title('Salary VS Experience(Training Set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()



#Visualising the test set 
plt.scatter(X_test, Y_test, color = 'red' )
plt.plot(X_train, regressor.predict(X_train), color = 'green' )
plt.title('Salary VS Experience(Test Set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()









